docs/import/color_function_util.js

```diff
@@ -25,22 +191,149 @@ function withColor(td, layer, property, index) {
       td.append(createColorBox(colorFunction['color']));
       break;
     case ColorStyle.DISCRETE:
-      const colorObject = colorFunction['colors'];
-      const colorSet = new Set();
-      Object.keys(colorObject).forEach((propertyValue) => {
-        const color = colorObject[propertyValue];
-        if (!colorSet.has(color)) {
-          colorSet.add(color);
-          td.append(createColorBox(colorObject[propertyValue]));
-        }
-      });
+      createColorBoxesForDiscrete(colorFunction, td);
       break;
     default:
       setStatus(ILLEGAL_STATE_ERR + 'unrecognized color function: ' + layer);
   }
+  td.addClass('editable')
+      .on('click', () => onClick(td, colorFunction['current-style']));
   return td;
 }
 
+/**
+ * On click function for color tds - updates the globalTd.
+ * @param {JQuery<HTMLElement>} td
+ * @param {enum} type
+ */
+function onClick(td, type) {
+  if ($(td).hasClass('na')) {
+    return;
+  }
+  const colorFunctionDiv = $('#color-fxn-editor');
+  if (colorFunctionDiv.is(':visible') && td === globalTd) {
+    colorFunctionDiv.hide();
+    $(globalTd).removeClass('selected');
+    return;
+  }
+  colorFunctionDiv.show();
+  if (td === globalTd) {
+    $(globalTd).addClass('selected');
+    return;
+  }
+  $(globalTd).removeClass('selected');
+  globalTd = td;
+  $(globalTd).addClass('selected');
+  $('#' + colorStyleTypeStrings.get(type) + '-radio')
+      .prop('checked', true)
+      .trigger('change');
```

Comment:
This trigger ends up writing to Firestore, right? What do you think of a method displaySchema(type) that has all of setSchema except for the last two lines. Then switchSchema can call it, and here, instead of triggering a change, you can call displaySchema. That way there's no writing to Firestore on clicking into the color.

Reply:
Done.